common/src/main/java/org/conscrypt/OpenSSLECGroupContext.java

```diff
@@ -24,22 +24,35 @@
 import java.security.spec.ECParameterSpec;
 import java.security.spec.ECPoint;
 import java.security.spec.EllipticCurve;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * Represents a BoringSSL EC_GROUP object.
  */
 final class OpenSSLECGroupContext {
+    private static final Map<String, String> ALIASES = new HashMap<>();
+    static {
+        // Workaround for OpenSSL not supporting SECG names for NIST P-256 (aka
+        // ANSI X9.62 prime256v1).
+        ALIASES.put("secp256r1", "prime256v1");
+        // OpenJDK uses string OIDs as names in lots of places, and expects them to
+        // be accepted in ECGenParameterSpec as well, so accept those
+        ALIASES.put("1.3.132.0.33", "secp224r1");
+        ALIASES.put("1.3.132.0.34", "secp384r1");
+        ALIASES.put("1.3.132.0.35", "secp521r1");
+        ALIASES.put("1.2.840.10045.3.1.7", "prime256r1");
```

Comment:
This appears to defeat the workaround mentioned above. Also there is no test for this slightly more complicated case below.

Reply:
I'm not sure I understand.  Both "secp256r1" and "1.2.840.10045.3.1.7" get replaced by "prime256r1", which is the curve name that BoringSSL understands.  (secp256r1 doesn't have its own OID.)

Reply:
The one above is prime256**v**1 and not with an "r".

Reply:
Ah, gotcha.  That's just a typo from typing too many secp curves.  Fixed that and added a test that all our curves work.